utils/scm/scmutils.go

```diff
@@ -0,0 +1,152 @@
+package scm
+
+import (
+	"errors"
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"github.com/jfrog/jfrog-client-go/xsc/services"
+)
+
+const (
+	Github        ScProvider = "github"
+	Gitlab        ScProvider = "gitlab"
+	Bitbucket     ScProvider = "bitbucket"
+	Azure         ScProvider = "azure"
+	Gerrit        ScProvider = "gerrit"
+	Gitea         ScProvider = "gitea"
+	AWSCodeCommit ScProvider = "codecommit"
+
+	Unknown ScProvider = ""
+)
+
+// ScProvider is the type of source control provider
+type ScProvider string
+
+func (sp ScProvider) String() string {
+	return string(sp)
+}
+
+const (
+	Git ScType = "git"
+)
+
+// ScType is the type of source control manager
+type ScType string
+
+func (st ScType) String() string {
+	return string(st)
+}
+
+// ScmTypeData holds the data for each source control manager type
+type ScmTypeData struct {
```

Comment:
I think the term scmTypeData is a bit confusing. It took me a second to realize you were referring to the indicator. Maybe we should call it ScmTypeIndicator?